src/ceph-disk/ceph_disk/main.py

```diff
@@ -122,7 +122,7 @@ def get_ready_by_type(what):
 
     @staticmethod
     def get_ready_by_name(name):
-        return [x[name]['ready'] for x in PTYPE.values()]
+        return [x[name]['ready'] for x in PTYPE.values() if x.get(name, None)]
```

Comment:
why not just let it throw if the `name` is not in `x`? as it is supposed to be in `x`.